### DISCUSSION TITLE
Any way to achieve this?

### DISCUSSION THREAD

**Discussion by xfzv** [tomasklaen/uosc#308] [Q&A] [answered]
I'd like to execute `cycle pause ; script-binding uosc/flash-pause-indicator` for short right click and `script-binding uosc/menu` for longer right click.

I tried with `MBTN_LEFT script-binding uosc/menu` added to `input.conf` and

```
 # Execute command for background clicks shorter than this number of milliseconds, 0 to disable 
 click_threshold=400
 click_command=cycle pause; script-binding uosc/flash-pause-indicator 
```

but it doesn't work. uosc menu is opened regardless of the left click duration.

Is this already possible and I'm missing something or could it be implemented in the future?

**Comment by tomasklaen** [comment, marked as answer]
`click_threshold` is for mouse left (primary) click, to provide an interface where people can both drag the window and play/pause the video with left mouse button as that is an expected behavior that is common for other players/apps. If you enable it, uosc will handle `MBTN_LEFT` internally based on `click_threshold` and `click_command` options. You shouldn't also set `MBTN_LEFT` in `input.conf` as that is going to have a priority and overwrite uosc's handling.

If you want to make your **right** (or any other) click execute different commands based on how long it was held down, you have to write your own script. I'll definitely not be implementing it, as that is a horrible UX. Why would you want to hold a button for alternative action without indication of when it kicks in, so you have to guess and hold for long enough to be sure you are getting the 2nd action ... this just sounds horrible to use. It's way better to bind stuff to your mouse wheel click, or back/forward buttons, or use modifiers.

**Reply by xfzv** [reply]
Just found [christoph-heinrich/mpv-pointer-event](https://github.com/christoph-heinrich/mpv-pointer-event), if someone is interested.

**Reply by christoph-heinrich** [reply]
> I'll definitely not be implementing it, as that is a horrible UX

Differentiating between short and long click is actually pretty common for touch devices.

**Reply by tomasklaen** [reply]
Yes, if you have an indicator of passing the threshold. None of the devices will make you just guess when that happens. This crucial difference is what makes it horrible.

And I'd say implementing that is outside of the scope of uosc.

**Reply by FichteFoll** [reply]
Completely off-topic:

> if you have an indicator of passing the threshold

I love how recent-ish android versions (11&12) at least completely forgot about this when previous versions supported that. Long-touching an input field would briefly vibrate my phone to indicate I passed the threshold for the long-touch operation (which then opens a context menu of sorts), while with 11 I don't get that at all and have to guess whether I passed it or not.

Other than that, a good indictor is just performing the long-touch operation as soon as the threshold is met, assuming it is actually visible enough and not obscured by your finger.

**Reply by christoph-heinrich** [reply]
> a good indictor is just performing the long-touch operation as soon as the threshold is met

That's what I'm doing in [pointer-event](https://github.com/christoph-heinrich/mpv-pointer-event). Why one would do it any other way is beyond me. In android it's inconsistent. In some places it performs the action as soon as the long touch time is reached, and in others it waits until the finger is released.
